lib/async_hooks.js

```diff
@@ -281,7 +278,8 @@ class AsyncLocalStorage {
 
   exit(callback, ...args) {
     process.nextTick(() => {
-      this.disable();
+      const resource = executionAsyncResource();
+      resource[this.kResourceStore] = undefined;
```

Comment:
It is almost the same as the previous exit (without the check on resource) but as duplicated code now

Reply:
Yeah, only needs to happen during disables now though, so still worthwhile. Maybe could still use the _exit for shared logic, but it at least eliminates the `enabled` property and makes the scoping more correct with the `AsyncResource`.

Reply:
Restored the `_exit` method.